app.js

```diff
@@ -165,88 +165,111 @@ app.get("/login", routes.login);
 // resource proxying for http-on-https
 webmakerProxy(app, middleware.checkForAuth);
 
-// Entry point for authenticated users
+/*
+ * Main routes for thimble.webmaker.org
+ */
+// Entry point for all users
 app.get('/',
+        middleware.setUserIfTokenExists,
+        middleware.setPublishUser,
+        routes.main.root);
+
+// Main route for authenticated users
+app.get('/user/:username/:projectId',
         middleware.redirectAnonymousUsers,
-        middleware.setNewPageOperation,
         middleware.setUserIfTokenExists,
         middleware.setPublishUser,
-        routes.root);
+        middleware.setProject,
+        routes.main.authenticated);
 
-// Entry point for anonymous users
+// Main route for anonymous users
 app.get('/anonymous/:anonymousId/:remixId?',
         middleware.redirectAuthenticatedUsers,
-        routes.main);
+        routes.main.anonymous);
 
-app.get('/getFileContents/:remixId?',
-        middleware.setUserIfTokenExists,
-        routes.getFileContents);
-
-app.get('/getFileMeta/:remixId?',
-        middleware.setUserIfTokenExists,
-        routes.getFileMetadata);
 
+/*
```

Comment:
These separations would be great as separate files. What are the chances you could factor them out to clean up this file?